Example/Tests/PKCS8Tests.swift

```diff
@@ -32,4 +33,121 @@ final class PKCS8Tests: XCTestCase {
     // then
     XCTAssertNil(convertedPKCS8PEM)
   }
+
+  func testEncryptPKCS1_simplePassword__shouldEncrypt() {
+    // given
+    let pkcs1PEMData = TestData.openSSLPrivateKeyPEM.data
+    let password = "password"
+
+    // when
+    let encrypted = PKCS8.encrypt(pkcs1PEMData, password: password)
+
+    // then
+    XCTAssertNotNil(encrypted)
+  }
+
+  func testEncryptPKCS1_emptyPassword__shouldReturnNil() {
+    // given
+    let pkcs1PEMData = TestData.openSSLPrivateKeyPEM.data
+    let password = ""
+
+    // when
+    let encrypted = PKCS8.encrypt(pkcs1PEMData, password: password)
+
+    // then
+    XCTAssertNil(encrypted)
+  }
+
+  func testEncryptPKCS1_uuidPassword__shouldEncrypt() {
+    // given
+    let pkcs1PEMData = TestData.openSSLPrivateKeyPEM.data
+    let password = UUID().uuidString
+
+    // when
+    let encrypted = PKCS8.encrypt(pkcs1PEMData, password: password)
+
+    // then
+    XCTAssertNotNil(encrypted)
+  }
+
+  func testEncryptPKCS1_sha512Password__shouldEncrypt() {
+    // given
+    let pkcs1PEMData = TestData.openSSLPrivateKeyPEM.data
+    let password = "B109F3BBBC244EB82441917ED06D618B9008DD09B3BEFD1B5E07394C706A8BB980B1D7785E5976EC049B46DF5F1326AF5A2EA6D103FD07C95385FFAB0CACBC86"
+
+    // when
+    let encrypted = PKCS8.encrypt(pkcs1PEMData, password: password)
+
+    // then
+    XCTAssertNotNil(encrypted)
+  }
+
+  func testEncryptPKCS1_specialCharactersPassword__shouldEncrypt() {
```

Comment:
as we talked about, testing with C string termination (\0) could make sense as well.

Reply:
added